components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/AuthenticatorMgtExceptionBuilder.java

```diff
@@ -0,0 +1,148 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.util;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtClientException;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtServerException;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtServerRuntimeException;
+
+/**
+ * Utility class for building authenticator management exceptions.
+ */
+public class AuthenticatorMgtExceptionBuilder {
+
+    private AuthenticatorMgtExceptionBuilder() {
+
+    }
+
+    public static AuthenticatorMgtClientException buildClientException(AuthenticatorMgtError error, String... data) {
+
+        String description = error.getDescription();
+        if (ArrayUtils.isNotEmpty(data)) {
+            description = String.format(description, data);
+        }
+
+        return new AuthenticatorMgtClientException(error.getCode(), error.getMessage(), description);
+    }
+
+    public static AuthenticatorMgtServerException buildServerException(AuthenticatorMgtError error, String... data) {
+
+        String description = error.getDescription();
+        if (ArrayUtils.isNotEmpty(data)) {
+            description = String.format(description, data);
+        }
+
+        return new AuthenticatorMgtServerException(error.getCode(), error.getMessage(), description);
+    }
+
+    public static AuthenticatorMgtServerException buildServerException(AuthenticatorMgtError error, Throwable e,
+                                                                       String... data) {
+
+        String description = error.getDescription();
+        if (ArrayUtils.isNotEmpty(data)) {
+            description = String.format(description, data);
+        }
+
+        return new AuthenticatorMgtServerException(error.getCode(), error.getMessage(), description, e);
+    }
+
+    public static AuthenticatorMgtServerRuntimeException buildRuntimeServerException(AuthenticatorMgtError error,
+                Throwable e, String... data) {
+
+        String description = error.getDescription();
+        if (ArrayUtils.isNotEmpty(data)) {
+            description = String.format(description, data);
+        }
+
+        return new AuthenticatorMgtServerRuntimeException(error.getCode(), error.getMessage(), description, e);
+    }
+
+    /**
+     * Enum class to represent the rule metadata errors.
+     */
+    public enum AuthenticatorMgtError {
+
+        // Client errors.
+        ERROR_NOT_FOUND_AUTHENTICATOR("60001", "No Authenticator found.",
+                "No Authenticator found by given authenticator name: %s."),
+        ERROR_OP_ON_SYSTEM_AUTHENTICATOR("60002", "No operations allowed on system authenticators.",
```

Comment:
This error enum is bit hard to understand. I first thought it's a typo. Can we give a more readable name